File: Programmers/Lv0/겹치는선분의길이.cpp
#include <string>
#include <vector>
#include <map>
#include <algorithm>
using namespace std;

int solution(vector<vector<int>> lines) {
    vector<vector<int>> alline;
    int answer = 0;

    for (vector<int> line : lines)
    {
        for (int i = 0; i < line[1] - line[0]; i++) {
            vector<int> tmp;
            tmp.push_back(line[0] + i);
            tmp.push_back(line[0] + i + 1);
            alline.push_back(tmp);
        }
    }
    sort(alline.begin(), alline.end());
    map<vector<int>, int> _map;
    for (auto line : alline)
        _map[line]++;
    for (auto pair : _map)
        if (pair.second != 1)
            answer++;
    return answer;
}

File: Programmers/Lv0/겹치는선분의길이.py
def remove_all(lst, value):
    return [x for x in lst if x != value]

def solution(lines):
    alline = []
    answer = 0
    for line in lines:
        for i in range(line[1] - line[0]):
            alline.append([line[0] + i, line[0] + i + 1])

    for line in alline:
        if alline.count(line) > 1:
            answer += 1
            alline = remove_all(alline, line)

    return answer